src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -397,6 +405,62 @@ internal async Task CleanUpUnusedAutoIndexes(DatabaseObservationState databaseSt
             }
         }
 
+        internal async Task CleanUpCompareExchangeTombstones()
+        {
+            using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                foreach (var dbName in _server.Cluster.GetDatabaseNames(context))
+                {
+                    if (_server.Cluster.GetNumberOfCompareExchangeTombstones(context, dbName) > 0)
+                    {
+                        var maxEtag = GetMaxTombstonesEtagToDelete(context, dbName);
+                        if (maxEtag > 0)
+                        {
+                            var result = await _server.SendToLeaderAsync(new CleanCompareExchangeTombstonesCommand(dbName, maxEtag));
+                            await _server.Cluster.WaitForIndexNotification(result.Index);
+                        }
+                    }
+                }
+            }
+        }
+
+        private long GetMaxTombstonesEtagToDelete(TransactionOperationContext context, string dbName)
+        {
+            var dbRecord = _server.LoadDatabaseRecord(dbName, out _);
+
+            var maxEtag = long.MaxValue;
+
+            if (dbRecord.PeriodicBackups.Count == 0)
+                return 0;
```

Comment:
In this case, what _would_ delete them? Do we have code that don't create tombstones if there are no periodic deletions?

Reply:
done